### PR TITLE
Mob 7166 clean

### PR DESCRIPTION
**Pull request by SeanColombo**
https://wikia-inc.atlassian.net/browse/MOB-7166

AppPromoLanding screen. This is a tool that serves as a landing page for all wikis that have a mobile app (and will just show up as "not a page" for other wikis). This looks good on desktop and tablet, but anything narrower is irrelevant because mobile devices will automatically redirect to the store-page for the app on their platform.

### CODE REVIEW

**Comment by SeanColombo** [comment]
Handy links for testing:
http://starwars.sean.wikia-dev.com/wiki/Community_App
http://fallout.sean.wikia-dev.com/wiki/Community_App
http://walkingdead.sean.wikia-dev.com/wiki/Community_App
http://harrypotter.sean.wikia-dev.com/wiki/Community_App
http://muppets.sean.wikia-dev.com/wiki/Community_App
http://lyrics.sean.wikia-dev.com/wiki/Community_App <-- example of what it looks like on a wiki w/out one of our standard mobile apps

**Comment by SeanColombo** [comment]
Associated config PR: https://github.com/Wikia/config/pull/1895

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+	protected static $CACHE_KEY = "mobileAppConfigs";
+	protected static $CACHE_KEY_BRANCH = "branchioBranchKey";
+	protected static $CACHE_KEY_VERSION = 0.1;
+	protected static $CACHE_KEY_VERSION_BRANCH = 0.1;
+	protected static $CACHE_EXPIRY = 86400;
+
+	/**
+	 * Render HTML for whole App Promo Landing page.
+	 */
+	public function executeIndex() {
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: Old module-style naming conventions are deprecated in Nirvana

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+		wfProfileIn(__METHOD__);
+
+		// Since this "Community_App" article won't be found, we need to manually say it's okay so that it's not a 404.
+		$this->response->setCode(self::RESPONSE_OK);
+		
+		// Pull in the app-configuration (has data for all apps)
+		$appConfig = [];
+		$memcKey = wfMemcKey( static::$CACHE_KEY, static::$CACHE_KEY_VERSION );
```
Comment (TK-999) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: `$this->wg` is available here

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+		$response = F::app()->wg->memc->get( $memcKey );
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: `$this->wg` is available here

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+	protected static $CACHE_EXPIRY = 86400;
+
+	/**
+	 * Render HTML for whole App Promo Landing page.
+	 */
+	public function executeIndex() {
+		wfProfileIn(__METHOD__);
+
+		// Since this "Community_App" article won't be found, we need to manually say it's okay so that it's not a 404.
+		$this->response->setCode(self::RESPONSE_OK);
```
Comment (lizlux) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: There should be spaces inside parentheses, here and throughout the PR. 

`$this->response->setCode( self::RESPONSE_OK );`

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+		$srcs = AssetsManager::getInstance()->getGroupCommonURL( 'app_promo_landing_js' );
+		foreach( $srcs as $src ) {
+			$this->wg->Out->addScript( "<script type=\"{$wgJsMimeType}\" src=\"{$src}\"></script>" );
+		}
+
+		// Inject the JS
+		$srcs = AssetsManager::getInstance()->getGroupCommonURL( 'app_promo_landing_js' );
+		foreach( $srcs as $src ) {
+			$this->wg->Out->addScript( "<script type=\"{$wgJsMimeType}\" src=\"{$src}\"></script>" );
+		}
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php:76]: Duplicate?

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+		// render the custom App Promo Landing body (this includes the nav bar and the custom content).
+		$body = F::app()->renderView('AppPromoLanding', 'Content', [ ]);
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: `$this->app` is available in these functions

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+		OasisController::addBodyClass('oasis-one-column');
+
+		// adding 'appPromo' class as a CSS helper
+		OasisController::addBodyClass('appPromo');
+
+		// temporary grid transition code, remove after transition
+		OasisController::addBodyClass('WikiaGrid');
+
+		// render Oasis module to the 'html' var which the AppPromoLanding_index template will just dump out.
+		$this->html = F::app()->renderView( 'Oasis', 'Index', [ 'body' => $body ] );
```
Comment (TK-999) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: `$this->app` is available here

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php (hunk 4 above)
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: Should be spacey

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+ * a page with very-little chrome on it that will let a user send the app link to their phone.
+ *
+ * @author Sean Colombo
+ */
+class AppPromoLandingController extends WikiaController {
+	
+	private static $extraBodyClasses = []; // TODO: REMOVE - ONLY DURING THE TEMPORARY COPYING OF OASISCONTROLLER
+
+	const RESPONSE_OK = 200;
+	const APP_CONFIG_SERVICE_URL = "http://prod.deploypanel.service.sjc.consul/api/app-configuration/";
```
Comment (lizlux) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: use single quotes where the functionality is the same.

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+		$this->androidScreenShot = "http://wikia-mobile.nocookie.net/wikia-mobile/android-screenshots/{$this->config->app_tag}/1.png";
+		$this->androidStoreSrc = F::app()->wg->ExtensionsPath."/wikia/AppPromoLanding/images/playStoreButton.png";
+		
+		$this->iosPhoneSrc = F::app()->wg->ExtensionsPath."/wikia/AppPromoLanding/images/silverIphone.png";
+		$this->iosScreenShot = "http://wikia-mobile.nocookie.net/wikia-mobile/ios-screenshots/{$this->config->app_tag}/en/4.7/4.png.PNGCRUSH.png";
+		$this->iosStoreSrc = F::app()->wg->ExtensionsPath."/wikia/AppPromoLanding/images/appleAppStoreButton.png";
+
+		$this->imgSpacing = 1; // spacing between the image-grid cells.
+
+		wfProfileOut( __METHOD__ );
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php:215]: same for this method ^^ `$this->wg` and `$this->app` are available and `executeFoobar` naming style is deprecated

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+	 * a special page, we are doing this by stealing the "Community_App" article title.
+	 */
+	public static function onOutputPageBeforeHTML( OutputPage &$out, &$text ){
+		// Only steal this page if the wiki has an app configured.
+		$config = AppPromoLandingController::getConfigForWiki(F::app()->wg->CityId);
+		if($config !== null){
+			$title = $out->getTitle();
+			$origTitle = $title->getDBkey();
+			
+			if($origTitle == "Community_App"){
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: consider moving to constant?

File: extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php [hunk 11]
```diff
@@ -0,0 +1,380 @@
...
+<?= $debug ?>
+
+<div class='entireAppPromo'>
+	<div class='thumbsWrapper'>
+		<div class='thumbRow'>
+		<?php
+			$numThumbs = 0;
+			foreach($trendingArticles as $topArticle){
+				print "<div class='imageWrapper'>";
+					print "<img src='{$topArticle["imgUrl"]}' title='{$topArticle["title"]}' width='{$topArticle["width"]}' height='{$topArticle["height"]}'/>";
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php]: article title should be escaped

File: extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php [hunk 12]
```diff
@@ -0,0 +1,380 @@
...
+				} else if(($numThumbs % $numThumbsPerRow) == 0){
+					print "</div><div class='thumbRow'>";
+				}
+			}
+		?>
+		</div>
+		<div class='topContent'>
+			<div class='pitchOuter'>
+				<div class='pitchInner'>
+					<?= wfMsg( 'apppromolanding-pitch' ) ?>
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php]: should be `wfMessage(...)->escaped()` everywhere in the template, `wfMsg` is deprecated

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php [hunk 13]
```diff
@@ -0,0 +1,328 @@
...
+
+		// adding 'appPromo' class as a CSS helper
+		OasisController::addBodyClass('appPromo');
+
+		// temporary grid transition code, remove after transition
+		OasisController::addBodyClass('WikiaGrid');
+
+		// render Oasis module to the 'html' var which the AppPromoLanding_index template will just dump out.
+		$this->html = F::app()->renderView( 'Oasis', 'Index', [ 'body' => $body ] );
+
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php:92]: Do we want to allow user and site CSS and JS to load?

File: extensions/wikia/AppPromoLanding/AppPromoLanding.i18n.php [hunk 14]
```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+* Internationalisation file for the AppPromoLanding extension.
+*
+* @addtogroup Languages
+*/
+
+$messages = array();
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/AppPromoLanding.i18n.php]: Short array syntax should be used

Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/AppPromoLanding.i18n.php]: Fixed in next commmit.

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php (hunk 8 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: thanks.  :) fixed in next commit.

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php (hunk 1 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: Ah, oops.  What's the new naming convention?

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php (hunk 4 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: Hmm, yeah I have 'em squished all over. Made a pass through & spaced things out a bit.

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php (hunk 5 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php:76]: Completely. Good catch!

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php (hunk 10 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: Great call. In next commit.

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php (hunk 3 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php]: thanks. fixed in next commit.

File: extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php (hunk 5 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php:76]: I removed one in a commit that is now in the PR, but it detected this one as still being here, so this comment is still visible, but the problem is resolved :+1:

File: extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php (hunk 11 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php]: fixed in next commit. thanks :)

File: extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php (hunk 12 above)
Comment (SeanColombo) [inline extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php]: Updated everywhere in next commit. Thanks

File: extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php [hunk 15]
```diff
@@ -279,7 +279,7 @@
 			</div>
 			<div class='branchIoOuter'>
 				<div class='branchIoInner'>
-					<p><?= wfMessage( 'apppromolanding-custompitch', "<wbr><span>{$config->name}</span>" )->escaped() ?></p>
+					<p><?= wfMessage( 'apppromolanding-custompitch', "<wbr><span>{$config->name}</span>" )->plain() ?></p>
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php]: You can use `wfMessage(...)->rawParams( <html> )->escaped();`' :)

File: extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php [hunk 16]
```diff
@@ -0,0 +1,380 @@
...
+	<div class='belowThumbs'>
+		<a class='backLink' href='<?= $mainPageUrl ?>'>
+			<?= $larrSvgCode ?><?= wfMessage( 'apppromolanding-back' )->escaped() ?>
+		</a>
+		<div class='branchIo_alternate'>
+			<div class='branchIo'>
+				<div class='callToAction'><?= wfMessage( 'apppromolanding-call-to-action' )->escaped() ?></div>
+				<form id='branchIoForm' method='post' onsubmit='return sendSMS();'>
+					<input type='text' name='phoneNumber' placeholder='<?= wfMessage( 'apppromolanding-phone-num-placeholder' )->escaped() ?>'/><button
+						data-send="<?= htmlspecialchars( wfMessage( 'apppromolanding-button-get' )->escaped() ) ?>"
```
Comment (mszabo-wikia) [inline extensions/wikia/AppPromoLanding/templates/AppPromoLanding_Content.php]: `htmlspecialchars` is no longer needed

**Comment by SeanColombo** [comment]
@lizlux @TK-999 thanks for all of the help! :D  ...fixed the issues mentioned & tested that it's still co-operating on my devbox, and it looks like it's going smoothly.

**Comment by SeanColombo** [comment]
@TK-999 : does it look good to merge?

**Comment by mszabo-wikia** [comment]
LGTM

**Pull request merged**